File: CPP/Binary Tree BFS/Kth Largest Sum in a Binary Tree.cpp
class Solution {
public:
    long long kthLargestLevelSum(TreeNode* root, int k) {
        priority_queue<long>pq;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty()){
            long size=q.size();
            long sum=0;
            for(int i=0;i<size;i++){
                TreeNode* ele=q.front();
                q.pop();
                sum=sum+ele->val;
                // Push left and right children if they exist
                if(ele->left)q.push(ele->left);
                if(ele->right)q.push(ele->right);
            }
            pq.push(sum);
        }
        if (pq.size() < k) return -1;
        for (int i = 0; i < k - 1; i++) pq.pop();
        return pq.top();
    }
};

File: Python/Binary Tree BFS/Kth Largest Sum in a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        pq=[]
        q=[]
        
        q.append(root)
        while len(q) !=0:
            size=len(q)
            sum=0
            for i in range(size):
                ele=q.pop(0)
                sum+=ele.val
                if ele.left:
                    q.append(ele.left)
                if ele.right:
                    q.append(ele.right)
            heapq.heappush(pq,-sum)
        if len(pq)<k:
            return -1
        i=0
        while i <k-1:
            heapq.heappop(pq)
            i+=1
        return -heapq.heappop(pq)
